syntax = "proto3";

service Calculator {
  rpc Calculate(stream Request) returns (stream Response) {}
}

message Request {
  int32 operand1 = 1;
  int32 operand2 = 2;
  string operation = 3;
}

message Response {
  int32 result = 1;
  string error_message = 2;
}
